lib/test.gi

```diff
@@ -12,25 +12,68 @@
 ##  end;
 
 InstallGlobalFunction(ParseTestInput, function(str, ignorecomments)
-  local lines, inp, pos, outp, ign, i;
+  local lines, inp, pos, outp, ign, commands, i, skipstate, checkifelseendif;
   lines := SplitString(str, "\n", "");
   inp := [];
   pos := [];
   outp := [];
   ign := [];
+  commands := [];
   i := 1;
+  checkifelseendif := l -> ForAny(["#@if","#@else","#@endif"], x -> StartsWith(l, x));
+  # skipstate represents the current status of '#@if/#@else/#@endif'
```

Comment:
Why "endif" when GAP is using "fi"? Is this modelled on the C preprocessor? But not every GAP user will be familiar with C, so I'd suggest switching to `#@fi`

Reply:
I just did it blindly in C style, but I agree, `#@fi` would be more GAPy